tensordot2.py

```diff
@@ -0,0 +1,213 @@
+# Copyright 2019 The TensorNetwork Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+# FIXME: This is a modified version of code from the TensorFlow project.
+#        Legal requirements for including this??
+
+"""A modified version of TensorFlow's tensordot operation."""
+
+from __future__ import absolute_import
+from __future__ import division
+from __future__ import print_function
+
+import numpy as np
+from six.moves import xrange  # pylint: disable=redefined-builtin
+
+import tensorflow as tf
+
+def tensordot(a, b, axes, name=None):
```

Comment:
This is really good! Can you include some benchmarks of both graph building and runtime of using this?

Reply:
I have some benchmarks now, but am not sure where to put them. Are you suggesting they should be listed in a docstring or comment?